src/extension.ts

```diff
@@ -59,33 +59,34 @@ async function initClient(serverInfo: server.ServerInfo): Promise<RSPClient> {
 }
 
 function registerCommands(commandHandler: CommandHandler, context: vscode.ExtensionContext) {
+    const errorMessage = 'Unable to %ACTION% the server: ';
     const newLocal = [
         vscode.commands.registerCommand('server.start',
-            context => executeCommand(commandHandler.startServer, commandHandler, 'run', context)),
+            context => executeCommand(commandHandler.startServer, commandHandler, 'run', context, errorMessage.replace('%ACTION%', 'start'))),
         vscode.commands.registerCommand('server.restart',
-            context => executeCommand(commandHandler.restartServer, commandHandler, context)),
+            context => executeCommand(commandHandler.restartServer, commandHandler, context, errorMessage.replace('%ACTION%', 'restart'))),
         vscode.commands.registerCommand('server.debug',
-            context => executeCommand(commandHandler.debugServer, commandHandler, context)),
+            context => executeCommand(commandHandler.debugServer, commandHandler, context, errorMessage.replace('%ACTION%', 'debug'))),
         vscode.commands.registerCommand('server.stop',
-            context => executeCommand(commandHandler.stopServer, commandHandler, context)),
+            context => executeCommand(commandHandler.stopServer, commandHandler, context, errorMessage.replace('%ACTION%', 'stop'))),
         vscode.commands.registerCommand('server.remove',
-            context => executeCommand(commandHandler.removeServer, commandHandler, context)),
+            context => executeCommand(commandHandler.removeServer, commandHandler, context, errorMessage.replace('%ACTION%', 'remove'))),
         vscode.commands.registerCommand('server.output',
-            context => executeCommand(commandHandler.showServerOutput, commandHandler, context)),
+            context => executeCommand(commandHandler.showServerOutput, commandHandler, context, '')),
```

Comment:
Shouldn't we add an error message here?

Reply:
I left this empty because there is no way it can fail. It just read from a Map in serverExplorer. It's going to fail only if the Map is not populated but in that case you could not call this action. btw we can add a message here easily too